### PR TITLE
Implement test HTTP server

### PR DESCRIPTION
**Pull request by JunielKatarn**
Port in-process programmable HTTP server for integration testing purposes.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3038)

### CODE REVIEW

File: vnext/Desktop.Test.DLL/React.Windows.Desktop.Test.x64.def [hunk 1]
```diff
@@ -12,3 +12,13 @@ EXPORTS
 ?SetOnMessage@WebSocketServer@Test@React@Microsoft@@QEAAX$$QEAV?$function@$$A6AXV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z@std@@@Z
 ?SetMessageFactory@WebSocketServer@Test@React@Microsoft@@QEAAX$$QEAV?$function@$$A6A?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@$$QEAV12@@Z@std@@@Z
 ?SetOnError@WebSocketServer@Test@React@Microsoft@@QEAAX$$QEAV?$function@$$A6AX$$QEAUError@IWebSocket@React@Microsoft@@@Z@std@@@Z
+
+; Microsoft::React::Test::HttpServer
```
Comment (Khalef1) [inline vnext/Desktop.Test.DLL/React.Windows.Desktop.Test.x64.def]: Several of those methods are not ABI safe. Do you intend to use this dll in devmain as well?

Comment (JunielKatarn) [reply vnext/Desktop.Test.DLL/React.Windows.Desktop.Test.x64.def]: Yes.
We should be fine as long as the symbols are updated, shouldn't we?

Comment (JunielKatarn) [reply vnext/Desktop.Test.DLL/React.Windows.Desktop.Test.x64.def]: Note, this DLL will only be used in test code.

Comment (Khalef1) [reply vnext/Desktop.Test.DLL/React.Windows.Desktop.Test.x64.def]: No it's not just a matter of having the symbols updated. The layout of the STL objects you have at the DLL boundary is not guaranteed across compiler versions.

What you have right now is fine in ISS, but for devmain, you will either want to change the signature to be ABI safe, or consume sources and build the binary in devmain.

Comment (JunielKatarn) [reply vnext/Desktop.Test.DLL/React.Windows.Desktop.Test.x64.def]: @aeulitz can you please comment?

**Pull request merged**